app/lib/three_scale/search.rb

```diff
@@ -184,8 +184,13 @@ def table_and_column(column)
 
   module Helpers
 
+    # Maximum page size for search results
     MAX_PER_PAGE = 20
 
+    # By default sphinx paginates search results, and the default page size is 20.
+    # Setting to a high value "disables" sphinx-based pagination to use will_paginate instead
+    MAX_SEARCH_PAGE_SIZE = 1_000_000
```

Comment:
I didn't get why we need to provide this huge value for the search 😅 
On a first look, it feels like we're telling the DB to bring us a lot of records at once, which could impact in performance.

**IF** the problem is that we don't know if there will be a next page after the `X` records we request, couldn't we fix this by requesting `X + 1` records? If there's one item after the ones we need, it'd mean there's a next page.

Reply:
Yeah, I agree, it's not ideal...

The first step is fetching the IDs from Sphinx search engine, this is where this big number is used - we need to know how many results are available (`total_entries`), for the pagination provided by [will_paginate](https://github.com/mislav/will_paginate) to show the total number of pages (`total_pages`). So knowing only whether there will be the next page is not enough.

What I am not sure about is whether all of the records will actually be queried from the DB by the list of IDs (search results), or whether `will_paginate` will actually prevent from that and only pull the corresponding page. It would be worth checking this, and if needed - it will need to be improved. But it will not belong to this PR anyway.

Reply:
I was checking that gem and saw that we can actually provide the [total_entries](https://github.com/mislav/will_paginate/blob/master/lib/will_paginate/active_record.rb#L148) value to `will_paginate`. In case it's not provided, will_paginate should [perform a count for us](https://github.com/mislav/will_paginate/blob/c732deeba65e709e6c5d223584e2fc4448a12334/lib/will_paginate/active_record.rb#L68-L79).
I really don't know the ins and outs of Sphinx so I'm not 100% sure in the accuracy of that comment 😬 Now, from the looks of it, it feels like we're paginating our search inside the `ProxyRuleQuery` and then again with `will_paginate` 🤔 
Also, +1 to call this an improvement and tackle it in a different PR :v:

Reply:
@thalesmiguel Nice research. But that's what we do already - we take `total_entries` value that Sphinx returns and pass it to `will_paginate`. We can't rely on `will_paginate` calcs, because the count doesn't come from our Rails DB, but rather Sphinx engine.
In order to get the actual number of search results we need to pass a big number to `per_page`. See https://freelancing-gods.com/thinking-sphinx/v5/searching.html#pagination

Reply:
I would suggest naming this constant `SPHINX_PAGE_SIZE_INFINITE` to make clear what it is used for.